src/steps/findings/index.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+} from '@jupiterone/integration-sdk-core';
+
+import { getOrCreateAPIClient } from '../../client';
+import { IntegrationConfig } from '../../config';
+import { Entities, Steps, Relationships } from '../constants';
+import {
+  createFindingEntity,
+  createSourceFindingRelationship,
+} from './converter';
+import { createDataSourceKey } from '../dataSource/converter';
+
+export async function fetchFindings({
+  instance,
+  jobState,
+  logger,
+}: IntegrationStepExecutionContext<IntegrationConfig>) {
+  const apiClient = getOrCreateAPIClient(instance.config, logger);
+
+  await apiClient.iterateFindings(async (finding) => {
+    const findingEntity = await jobState.addEntity(
+      createFindingEntity(finding),
+    );
+    const sourceEntity = await jobState.findEntity(
+      createDataSourceKey(finding['Data Source']),
+    );
+
+    if (findingEntity && sourceEntity) {
+      await jobState.addRelationship(
+        createSourceFindingRelationship(sourceEntity, findingEntity),
+      );
```

Comment:
Should we add a log when we fail to findEntity or create the relationship? Also slight simplification:
```suggestion
    const sourceEntity = await jobState.findEntity(
      createDataSourceKey(finding['Data Source']),
    );

    if (sourceEntity) {
      await jobState.addRelationship(
        createSourceFindingRelationship(sourceEntity, findingEntity),
      );
```